modules/govuk_elasticsearch/manifests/init.pp

```diff
@@ -106,13 +121,18 @@
   }
 
   elasticsearch::instance { $::fqdn:
+    notify        => Exec['secure_es_yaml'],
     config        => $instance_config_real,
     datadir       => '/mnt/elasticsearch',
     init_defaults => {
       'ES_HEAP_SIZE' => $heap_size,
     },
   }
 
+  exec { 'secure_es_yaml':
+    command => "chmod 0640 /etc/elasticsearch/${::fqdn}/elasticsearch.yml",
+  }
```

Comment:
I would rather not use exec either. The problem is that this resource is controlled in the vendor elasticsearch module, which I'd rather not modify without pushing the change upstream. We could submit a PR to the vendor and update this code if they accept?